src/main/java/io/axoniq/axonserver/connector/ResultStream.java

```diff
@@ -86,6 +87,16 @@
      */
     void onAvailable(Runnable callback);
 
+    /**
+     * Accepts a callback to be invoked when the very first element of the stream is consumed/peeked.
```

Comment:
I would make it really clear in the javadoc that this returns a Stream that will, when the first element is consumed from it, trigger a callback.